From ea564b7da2d90d2e2414ec454bb3a7ab7cf4d0cb Mon Sep 17 00:00:00 2001
From: Eric Sandeen <sandeen@redhat.com>
Date: Mon, 4 Aug 2014 18:58:41 -0400
Subject: [fs] btrfs: replace BUG in can_modify_feature

Message-id: <1407179109-13714-35-git-send-email-sandeen@redhat.com>
Patchwork-id: 86982
O-Subject: [RHEL7.1 PATCH 034/422] btrfs: replace BUG in can_modify_feature
Bugzilla: 1071179
RH-Acked-by: Zach Brown <zab@redhat.com>

From: David Sterba <dsterba@suse.cz>

commit cc37bb04201217b7acb11213e16cb5530c30da8f
Author: David Sterba <dsterba@suse.cz>
Date:   Tue Nov 19 13:36:21 2013 +0100

    btrfs: replace BUG in can_modify_feature

    We don't need to crash hard here, it's just reading a sysfs file. The
    values considered in switch are from a fixed set, the default case
    should not happen at all.

    Signed-off-by: David Sterba <dsterba@suse.cz>
    Signed-off-by: Josef Bacik <jbacik@fb.com>
    Signed-off-by: Chris Mason <clm@fb.com>
Signed-off-by: Eric Sandeen <sandeen@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/btrfs/sysfs.c b/fs/btrfs/sysfs.c
index 8fdc052..b535285 100644
--- a/fs/btrfs/sysfs.c
+++ b/fs/btrfs/sysfs.c
@@ -75,7 +75,9 @@ static int can_modify_feature(struct btrfs_feature_attr *fa)
   clear = BTRFS_FEATURE_INCOMPAT_SAFE_CLEAR;
   break;
  default:
-  BUG();
+  printk(KERN_WARNING "btrfs: sysfs: unknown feature set %d\n",
+    fa->feature_set);
+  return 0;
  }
 
  if (set & fa->feature_bit)
-- 
1.7.1